backend/composer/models.py

```diff
@@ -767,6 +786,19 @@ def save(self, *args, **kwargs):
             self.reference_uri = create_reference_uri(self.pk)
             self.save(update_fields=["reference_uri"])
 
+    def delete(self, user=None, *args, **kwargs):
+        if self.has_statement_been_exported:
+            if not user:
+                raise ValidationError("User is required to deprecate the statement before deletion.")
+
+            try:
+                self.deprecated(by=user)
```

Comment:
This is fine but my preference would be to use the already existent method do_transition instead of calling the transition directly.

Done here: https://github.com/MetaCell/sckan-composer/commit/c544b469b68150186b52bb5c4488a5da023456d2